test/concentrate/vehicle_position_test.exs

```diff
@@ -23,5 +58,268 @@ defmodule Concentrate.VehiclePositionTest do
+                %{id: "busloc", value: "busloc_route"},
+                %{id: "swiftly", value: "swiftly_route"}
+              ]
+            }
+          ]
+        )
+
+      assert Mergeable.merge(swiftly, non_swiftly) == expected
+      assert Mergeable.merge(non_swiftly, swiftly) == expected
+      assert Mergeable.merge(swiftly_later, non_swiftly) == expected_later
+      assert Mergeable.merge(non_swiftly, swiftly_later) == expected_later
+    end
+
+    test "merge/2 takes the other trip and route values if the swiftly values are nil" do
+      swiftly =
+        new(
+          last_updated: 1,
+          latitude: 1,
+          longitude: 1,
+          trip_id: nil,
+          route_id: nil,
+          sources: MapSet.new(["swiftly"])
+        )
+
+      non_swiftly =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "busloc_trip",
+          route_id: "busloc_route",
+          sources: MapSet.new(["busloc"])
+        )
+
+      expected =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "busloc_trip",
+          route_id: "busloc_route",
+          sources: MapSet.new(["busloc", "swiftly"]),
+          data_discrepancies: [
+            %DataDiscrepancy{
+              attribute: :trip_id,
+              sources: [
+                %{id: "swiftly", value: nil},
+                %{id: "busloc", value: "busloc_trip"}
+              ]
+            },
+            %DataDiscrepancy{
+              attribute: :route_id,
+              sources: [
+                %{id: "swiftly", value: nil},
+                %{id: "busloc", value: "busloc_route"}
+              ]
+            }
+          ]
+        )
+
+      assert Mergeable.merge(swiftly, non_swiftly) == expected
+      assert Mergeable.merge(non_swiftly, swiftly) == expected
+    end
+
+    test "merge/2 defaults to the latest value if both come from swiftly" do
+      swiftly =
+        new(
+          last_updated: 1,
+          latitude: 1,
+          longitude: 1,
+          trip_id: "swiftly_trip",
+          route_id: "swiftly_route",
+          sources: MapSet.new(["swiftly"])
+        )
+
+      swiftly_merged =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "swiftly_merged_trip",
+          route_id: "swiftly_merged_route",
+          sources: MapSet.new(["busloc", "swiftly"])
+        )
+
+      expected =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "swiftly_merged_trip",
+          route_id: "swiftly_merged_route",
+          sources: MapSet.new(["busloc", "swiftly"]),
+          data_discrepancies: [
+            %DataDiscrepancy{
+              attribute: :trip_id,
+              sources: [
+                %{id: "swiftly", value: "swiftly_trip"},
+                %{id: "busloc|swiftly", value: "swiftly_merged_trip"}
```

Comment:
I don't understand how this situation arises: you shouldn't be merging already merged data.

Reply:
Removed